src/test/java/bio/terra/integration/DataRepoFixtures.java

```diff
@@ -852,12 +852,22 @@ private DataRepoResponse<SnapshotPreviewModel> retrieveSnapshotPreviewByIdRaw(
   public List<String> getRowIds(
       TestConfiguration.User user, DatasetModel dataset, String tableName, int limitRowsReturned)
       throws Exception {
+    return getColumnValues(user, dataset.getId(), tableName, PDAO_ROW_ID_COLUMN, limitRowsReturned);
+  }
+
+  public List<String> getColumnValues(
+      TestConfiguration.User user,
+      UUID datasetId,
+      String tableName,
+      String columnName,
+      int limitRowsReturned)
+      throws Exception {
     List<Object> dataModel =
-        retrieveDatasetData(user, dataset.getId(), tableName, 0, limitRowsReturned, null)
-            .getResult();
+        retrieveDatasetData(user, datasetId, tableName, 0, limitRowsReturned, null).getResult();
     assertThat("got right num of row ids back", dataModel.size(), equalTo(limitRowsReturned));
     return dataModel.stream()
-        .map(r -> ((LinkedHashMap) r).get(PDAO_ROW_ID_COLUMN).toString())
+        .filter(r -> ((LinkedHashMap) r).get(columnName) != null)
+        .map(r -> ((LinkedHashMap) r).get(columnName).toString())
         .toList();
```

Comment:
To avoid doing the cast twice, you can filter on the result of `get()`:
```
    //noinspection unchecked
    return dataModel.stream()
        .map(r -> ((LinkedHashMap<String, Object>) r).get(PDAO_ROW_ID_COLUMN))
        .filter(Objects::nonNull)
        .map(Object::toString)
        .toList();
```

Reply:
Thanks!